### DISCUSSION TITLE
PuppyRaflle Test

### DISCUSSION THREAD

**Discussion by guyzilberblum** [Cyfrin/security-and-auditing-full-course-s23#172] [Q&A]
I found a problem in the puppy raflle contract , at the refund function they dont delete the player they just make him the address(0)
my argument is that for iterating the array without deleting the players at the refund function will be gas costly so i wanted to wrtie a test case for that i had hard time doing that so i manage to log the gas for itterating 100 players and the log the gas for deleting them my qeastion is this the write proof to write ? this is the proof 
""" 

       function testBlindSpotCost () public {
        vm.txGasPrice(1);
        uint256 playersnumm = 100;
        address[] memory players = new address[](playersnumm);
        for (uint256 i = 0; i < playersnumm; i++) {
            players[i] = address(i);
        }

        uint256 gasStartt = gasleft();
        puppyRaffle.enterRaffle{value: entranceFee * playersnumm}(players);
        uint256 gasEndd = gasleft();
        uint256 gasUsedFirstt = (gasStartt - gasEndd) * tx.gasprice;
        console.log("Gas cost of itteration with blindspots", gasUsedFirstt);

        uint256 gasBefore = gasleft();
        vm.warp(block.timestamp + duration + 1);
        vm.roll(block.number + 1);
        puppyRaffle.selectWinner();
        uint256 gasAfter = gasleft();
        uint256 gasUsed = gasBefore - gasAfter;
         console.log("Gas used for deleting players array:", gasUsed);
    }
"""

**Comment by EngrPips** [comment]
Hello @guyzilberblum, What you wrote seems like a correct proof of code. That may pass for a `Gas Optimization` report. It's interesting that you are crafting these tests yourself now.

**Comment by guyzilberblum** [comment]
Step by step

**Reply by EngrPips** [reply]
Indeed.
